lib/stripe/api.ex

```diff
@@ -88,6 +100,16 @@ defmodule Stripe.API do
     end
   end
 
+  @spec add_idempotency_headers(headers, method) :: headers
+  defp add_idempotency_headers(existing_headers, method) when method in [:get, :head] do
+    existing_headers
+  end
+
+  defp add_idempotency_headers(existing_headers, _method) do
+    existing_headers
+    |> Map.put(@idempotency_key_header, generate_idempotency_key())
```

Comment:
Should this not be `Map.put_new/2? This to allow the user to provide their own idempotency key that won't be overriden with the generated key.

Reply:
I never thought about that until you mentioned it 😄 Thanks! I'll fix it later.

Reply:
@maartenvanvliet Fixed in bc7e57c